### PR TITLE
eless 0.5 (new formula)

### PR DESCRIPTION
**Pull request by raxod502**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

(Well, `brew audit --strict` complains that the repository has only 49 stars instead of 50, but that seems a bit nitpicky to me.)

-----

See also https://github.com/kaushalmodi/eless/issues/30#issuecomment-368092548.

### CODE REVIEW

**Comment by kaushalmodi** [comment]
It's now 50 stars. What's next?

File: Formula/eless.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class Eless < Formula
+  desc "Better 'less' using Emacs view-mode and Bash"
+  homepage "https://eless.scripter.co"
+  url "https://github.com/kaushalmodi/eless/archive/v0.5.tar.gz"
+  sha256 "b4da2c7c223996681bb951d50e0d542d8df04baf765c16412fa0848cdb2b3a3d"
+
+  head do
```
Comment (DomT4) [inline Formula/eless.rb]: Kill the block and just use `head "https://github.com/kaushalmodi/eless.git"` under `sha256` above.

File: Formula/eless.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    url "https://github.com/kaushalmodi/eless.git"
+  end
+
+  depends_on "emacs"
+  depends_on "bash"
```
Comment (DomT4) [inline Formula/eless.rb]: Both `bash` and `perl` are provided by the system. Are there actually benefits to forcing the latest/greatest versions here?

Comment (kaushalmodi) [reply Formula/eless.rb]: Hello, I am the eless utility developer. It's tested to work in Perl 5.16.3 (RHEL 6.6, my machine) and bash 3.2 (as reported by a macOS user). If those versions are old enough for a majority of systems, I believe that enforcing of the latest versions can be skipped.

File: Formula/eless.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  depends_on "emacs"
+
+  def install
+    bin.install "eless"
+    info.install "docs/eless.info"
+  end
+
+  test do
+    system bin/"eless", "-V"
```
Comment (DomT4) [inline Formula/eless.rb]: Can this test be made to exercise functionality on a slightly deeper level at all?

**Comment by kaushalmodi** [comment]
I have a full test suite on the project repo: https://github.com/kaushalmodi/eless/blob/master/Makefile

I am not macOS user .. how would you suggest translate that "make test" to a test in this scenario? 

Eless is an interactive utility, so it's a bit challenging to test without going through the method I went in that "make test".

**Comment by raxod502** [comment]
@kaushalmodi 

I tried

```ruby
  test do
    ENV["HOME"] = testpath
    mkdir testpath/".emacs.d"
    (testpath/".emacs.d"/"elesscfg").write(<<~EOS
      (write-file "eless_output")
      (kill-emacs)
      EOS
                                          )
    (testpath/"eless_input").write("hello")
    system "#{bin}/eless", testpath/"eless_input"
    assert_equal "hello", (testpath/"eless_output").read
  end
```

But this gave me the following error message from `eless`:

```
This script is not supposed to send output to a pipe
```

What do you think?

**Comment by kaushalmodi** [comment]
> What do you think?

I don't know the language you used to write that script, but I can sort of understand it.. looks like a light-weight version of https://github.com/kaushalmodi/eless/blob/master/test/elesscfg_for_tests, and it's not piping eless output to another process.. so it's odd to see that error assuming that the `system "#{bin}/eless", testpath/"eless_input"` syntax doing something like `eless "${testpath}/eless_input"` in bash.

**Comment by raxod502** [comment]
@DomT4 does Homebrew implicitly capture stdout for `system` commands? Is there a way to disable that?

**Comment by raxod502** [comment]
Any update here?

**Comment by DomT4** [comment]
https://github.com/Homebrew/brew/blob/057d3461a9c0693b35d19305677cb4866155acf4/Library/Homebrew/utils.rb#L182-L200

**Comment by raxod502** [comment]
@DomT4 Although I can't see how the code you linked is relevant to my question, it seems on further investigation that Homebrew does indeed implicitly capture stdout. I'm assuming there's no way to change this, so we will either have to make `eless` run without being connected to a TTY or accept that the test can't be improved.

@kaushalmodi Do you think it is possible for `eless` to operate without a TTY, for example by providing a way to run Emacs in `--batch` mode?

**Comment by kaushalmodi** [comment]
@raxod502 

TTY is at the core of eless functionality:

https://github.com/kaushalmodi/eless/blob/530353ef26963aeadb6afbb67e99bcb5059cb827/eless#L512

It's an interactive use utility.. The only non-interactive "batch mode" eless functionality is showing the version (`eless -V`) and showing help (`eless -h`).

I hope they are pass..

**Comment by raxod502** [comment]
@DomT4 Sounds like we can't improve the test, unfortunately. Any other changes that need to be made to the formula before merging?

**Review by DomT4 — APPROVED** [review]

**Comment by DomT4** [comment]
Dropping my objection here. It'd be nice to have a `makefile` that can do `make install prefix=BLAH` rather than manually placing stuff around but I'll leave that one to ILZ.

**Comment by commitay** [comment]
> It'd be nice to have a `makefile` that can do `make install prefix=BLAH` rather than manually placing stuff

@kaushalmodi Is it possible to have an install script for this?

**Comment by kaushalmodi** [comment]
@commitay It's possible. But I'll need to dig into learning canonical way to write Makefiles. I didn't do it in the first place as I fear I might end up messing up someone's files and folders while trying to install this small script. If you have pointer to a good Makefile, let me know. I'll find some time for this within a week.

**Comment by ilovezfs** [comment]
An `install.sh` would be preferable to a `Makefile` so we can keep it bottle unneeded. See
https://github.com/Homebrew/homebrew-core/blob/master/Formula/bats.rb#L11
https://github.com/Homebrew/homebrew-core/blob/master/Formula/bash-snippets.rb#L38
https://github.com/Homebrew/homebrew-core/blob/master/Formula/clojure.rb#L12-L14 
https://github.com/Homebrew/homebrew-core/blob/master/Formula/tmux-xpanes.rb#L12
etc.

**Comment by kaushalmodi** [comment]
Please keep this open. I haven't yet got a chance to look at this. I'll make sure to add that install script over the weekend.

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by ilovezfs** [comment]
Any update here on the

a) install script and

b) test of functionality

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by kaushalmodi** [comment]
@ilovezfs Sorry, I haven't yet got a chance to update this.

**Comment by ilovezfs** [comment]
@kaushalmodi that's OK. We'll go with this as-is. An install script would be great though.

**Comment by ilovezfs** [comment]
Thanks for the 🆕 formula and another contribution to Homebrew @raxod502! 🚀

**Comment by kaushalmodi** [comment]
@raxod502 Thanks for your contribution. Apologies for the inconvenience created.

**Pull request merged**